torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -750,14 +750,12 @@ def pad_bounding_box(
 
     left, right, top, bottom = _parse_pad_padding(padding)
 
-    bounding_box = bounding_box.clone()
-
     # this works without conversion since padding only affects xy coordinates
-    bounding_box[..., 0] += left
-    bounding_box[..., 1] += top
     if format == features.BoundingBoxFormat.XYXY:
-        bounding_box[..., 2] += left
-        bounding_box[..., 3] += top
+        pad = torch.tensor([left, top, left, top], device=bounding_box.device)
```

Comment:
Should we also set the `dtype` here? Otherwise `pad` will always by `int32`. Depending on the dtype of `bounding_box` two issues can happen here:

1. If the bounding box is floating point or int64, the `+` later on will convert it to the correct type, but this is a wasted conversion. Admittedly, this can hardly regarded as perf improvement since we are talking about tiny tensors here.
2. If the bounding box is of a "smaller" integer type, e.g. int16, the `+` later on will upcast it to `int32` and thus violate our rule that the kernel should retain the dtype. This is not caught by the test suite, because we are only testing 
   https://github.com/pytorch/vision/blob/e64784cdea465d833d9d0f66dc73d7abe217933d/test/prototype_common_utils.py#L397

Reply:
The same also applies to `crop_bounding_box` that was changed in #6881 and is also touched in this PR.

Reply:
Yes, I was thinking also about dtype. In general, yes, we need to convert pad to bounding_box dtype here, in practice my thought was that bboxes are floats so padding will be converted to float. I think bboxes as int16 are rare...
OK, i'll update the PR.